### DISCUSSION TITLE
Best Configuration

### DISCUSSION THREAD

**Discussion by tonytorm** [ggml-org/whisper.cpp#1608] [Q&A]
Been busy but finally managed to download the laster updates from whisper.cpp and looking for better performance, 
and honestly don't understand where to find the best performance, at the mo i have GGML_USE_ACCELERATE and GGML_USE_METAL both on but i don't see a huge difference with older versions of whisper.cpp, especially with bigger models and especially the METAL usage seems to make almost no difference, am I overlooking something?

**Comment by LVCSRer** [comment]
I tested it on the iPhone, focusing on processing speed.

On iPhone 12, I tested 3 seconds and 15 seconds input, and Core ML without Metal was best for test device. As the input length increases, the decode time with Metal increases rapidly.

Below are my test result. (Sorry don't know how to use table for markup)


hello how are you (3s) | with Metal												
											the last column for average
mel	12	14	12	14	13	14	13	14	13	13	13.2	
sample	10	12	13	14	12	12	12	13	14	13	12.5	
encode	343	233	234	232	232	232	233	232	233	242	244.6	
decode	72	68	65	67	68	69	69	68	69	67	68.2	
batchd	78	22	23	23	23	23	22	23	23	23	28.3	
prompt	0											
Total	515	349	347	350	348	350	349	350	352	358	366.8	



hello how are you (3s) | without Metal 												
										
mel	14	14	13	12	14	14	14	14	13	13.55555556		
sample	3	3	3	3	3	3	3	3	3	3		
encode	361	247	252	252	249	243	248	253	249	261.5555556		
decode	41	39	38	38	39	38	39	38	38	38.66666667		
batchd	33	30	29	32	27	32	29	31	27	30		
prompt												
Total	452	333	335	337	332	330	333	339	330	346.7777778		
												
												

15s | with Metal												
											
mel	27	28	27	27	27	27	27	27	28	27	27.2	
sample	74	97	100	103	108	111	103	100	104	100	100	
encode	489	451	447	447	440	436	444	448	449	439	449	
decode	620	687	659	675	693	695	693	675	705	673	677.5	
batchd	37	22	22	22	22	22	22	23	22	22	23.6	
prompt												
Total	1247	1285	1255	1274	1290	1291	1289	1273	1308	1261	1277.3	



15s | without Metal												
												
mel	27	27	28	27	27	27	27	38	28	28	28	28.5
sample	21	21	21	21	21	21	21	22	21	21	21	21.1
encode	546	477	469	466	472	468	480	478	497	479	481	476.7
decode	266	257	259	259	260	257	257	270	261	257	260	259.7
batchd	29	28	31	33	29	34	27	29	29	26	30	29.6
prompt												
Total	889	810	808	806	809	807	812	837	836	811	820	815.6

**Reply by LVCSRer** [reply]
Hope my question also answered :) 
https://github.com/ggerganov/whisper.cpp/discussions/1722
